lib/atat-web-api-stack.ts

```diff
@@ -3,22 +3,32 @@ import * as apigw from "aws-cdk-lib/aws-apigateway";
 import * as iam from "aws-cdk-lib/aws-iam";
 import * as statement from "cdk-iam-floyd";
 import { Construct } from "constructs";
-import { AtatRestApi } from "./constructs/apigateway";
+import { AtatNetStack } from "./atat-net-stack";
+import { AtatRestApi, AtatRestApiProps } from "./constructs/apigateway";
 import { UserPermissionBoundary } from "./aspects/user-only-permission-boundary";
 import { ApiSfnFunction } from "./constructs/api-sfn-function";
 import { HttpMethod } from "./http";
 import { ProvisioningWorkflow } from "./constructs/provisioning-sfn-workflow";
 
 export interface AtatWebApiStackProps extends cdk.StackProps {
   environmentName: string;
+  network?: AtatNetStack;
 }
 
 export class AtatWebApiStack extends cdk.Stack {
   constructor(scope: Construct, id: string, props: AtatWebApiStackProps) {
     super(scope, id, props);
-    const { environmentName } = props;
+    const { environmentName, network } = props;
 
-    const api = new AtatRestApi(this, "SampleApi");
+    const apiProps: AtatRestApiProps = {};
+    if (network) {
+      apiProps.vpcConfig = {
+        vpc: network.vpc,
+        interfaceEndpoint: network.endpoints.apigateway,
+      };
+    }
```

Comment:
Should we throw an error here if `network` isn't provided?

Reply:
We shouldn't. The better solution if we wanted to require `network` would to be to just... require `network` :) See https://github.com/dod-ccpo/atat-web-api/pull/739#discussion_r844507277 for more information.